core/primitives/sector_file/sector_file.cpp

```diff
@@ -454,6 +459,44 @@ namespace fc::primitives::sector_file {
     };
   }
 
+  outcome::result<boost::optional<PieceInfo>> SectorFile::writeNull(
+      PaddedByteIndex offset,
+      PaddedPieceSize size,
+      const boost::optional<RegisteredSealProof> &maybe_seal_proof_type) {
+    if (not file_.good()) {
+      return SectorFileError::kInvalidFile;
+    }
+
+    const auto piece = runsAnd(runs_, std::vector<uint64_t>({offset, size}));
+    OUTCOME_TRY(allocated_size, runsCount(piece));
+
+    if (allocated_size > 0) {
+      logger_->warn(
+          "getting partial file writer overwriting {} allocated bytes",
+          allocated_size);
+    }
+
+    file_.seekp(offset, std::ios_base::beg);
```

Comment:
Convertion from unsigned to signed, uses `gsl::narrow`
```suggestion
    file_.seekp(gsl::narrow<int64_t>(offset), std::ios_base::beg);
```

Reply:
Fixed